dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/subcommands/impl/All.java

```diff
@@ -118,43 +147,89 @@ protected void forkRepositoriesFound(Map<String, String> parentToPath,
+                return;
+            }
         } else {
             return;
         }
-        GHRepository parent = retrievedRepo.getParent();
+        GHRepository parent = forkedRepo.getParent();
 
-        if (parent == null || !parentToPath.containsKey(parent.getFullName())) {
+        if (parent == null || !pathToDockerfilesInParentRepo.containsKey(parent.getFullName())) {
             return;
         }
-        log.info("Fixing Dockerfiles in {}...", initialRepo.getFullName());
-        String parentName = parent.getFullName();
-        String image = parentToImage.get(parentName);
-        String tag = imageToTagMap.get(image);
-        String branch = retrievedRepo.getDefaultBranch();
-        GHContent content = dockerfileGitHubUtil.tryRetrievingContent(retrievedRepo, parentToPath.get(parentName),
-                branch);
-        dockerfileGitHubUtil.modifyOnGithub(content, branch, image, tag, commitMessage);
-        dockerfileGitHubUtil.createPullReq(parent, branch, retrievedRepo, message);
-    }
 
+        log.info("Fixing Dockerfiles in {}...", forkedRepo.getFullName());
+        String parentName = parent.getFullName();
+        String branch = (ns.get("b") == null) ? forkedRepo.getDefaultBranch() : ns.get("b");
+
+        String pathToDockerfile;
+        String image;
+        String tag;
+        GHContent content;
+        boolean isContentModified = false;
+        boolean isRepoSkipped = true;
+
+        Iterator<String> pathToDockerfileInParentRepoIterator = pathToDockerfilesInParentRepo.get(parentName).iterator();
+        Iterator<String> imagesFoundInParentRepoIterator = imagesFoundInParentRepo.get(parentName).iterator();
+
+        while (pathToDockerfileInParentRepoIterator.hasNext() && imagesFoundInParentRepoIterator.hasNext()) {
+            pathToDockerfile = pathToDockerfileInParentRepoIterator.next();
+            image = imagesFoundInParentRepoIterator.next();
+            tag = imageToTagMap.get(image);
+            log.info("pathToDockerfile: {} , image: {}, tag: {}", pathToDockerfile, image, tag);
+            content = dockerfileGitHubUtil.tryRetrievingContent(forkedRepo, pathToDockerfile, branch);
+            log.info("content: {}", content);
+            if (content != null) {
+                dockerfileGitHubUtil.modifyOnGithub(content, branch, image, tag, ns.get("c"));
+                isContentModified = true;
+                isRepoSkipped = false;
+            } else {
+                log.info("No Dockerfile found at path: '{}'", pathToDockerfile);
+            }
+        }
 
+        if (isRepoSkipped) {
+            log.info("Skipping repo '{}' because contents of it's fork could not be retrieved. Moving ahead...",
+                    parentName);
+            skippedRepos.add(forkedRepo.getFullName());
+        }
 
+        if (isContentModified) {
+            dockerfileGitHubUtil.createPullReq(parent, branch, forkedRepo, ns.get("m"));
```

Comment:
You might want to create identifying constants for things getting checked in `ns.get`. E.g. `ns.get(Constants.GIT_REPO)`